icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitsData.java

```diff
@@ -0,0 +1,98 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html#License
+/*
+ *******************************************************************************
+ * Copyright (C) 2004-2020, Google Inc, International Business Machines
+ * Corporation and others. All Rights Reserved.
+ *******************************************************************************
+ */
+
+package com.ibm.icu.impl.units;
+
+import com.ibm.icu.impl.Assert;
+import com.ibm.icu.impl.ICUData;
+import com.ibm.icu.impl.ICUResourceBundle;
+import com.ibm.icu.impl.UResource;
+import com.ibm.icu.util.UResourceBundle;
+
+import java.util.ArrayList;
+
+/**
+ * Responsible for all units data operations (retriever, analysis, extraction certain data ... etc.).
+ */
+public class UnitsData {
```

Comment:
I'm not entirely convinced that SimpleUnitIdentifiersSink and these Constants belong in UnitsData.java.

@sffc do you know if there's similar need for the ability to keep things small in Java-land as in C++-land?

In C++, this code is still in measunit_extra.cpp. In theory people could use unit identifiers without using our new unit formatting pipeline code with conversions etc. I don't know if anyone would do that though... unitsdata.cpp/unitsdata.h are associated with the latter, and separate of the former. These

Reply:
lets discuss today
@sffc , @hugovdm